coral/src/app/components/AuthenticationRequiredBoundary.tsx

```diff
@@ -0,0 +1,43 @@
+import { Dialog } from "@aivenio/design-system";
+import { Component, ReactElement } from "react";
+import { isUnauthorizedError } from "src/services/api";
+
+class AuthenticationRequiredBoundary extends Component<{
+  children: ReactElement;
+}> {
+  public state = {
+    hasError: false,
+  };
+  constructor(props: { children: ReactElement }) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error: unknown) {
+    if (isUnauthorizedError(error)) {
+      return { hasError: true };
+    }
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <Dialog
+          title={"Authentication required"}
+          type={"warning"}
+          open={true}
+          primaryAction={{
+            onClick: () => (window.location.href = "/login"),
```

Comment:
Ah, probably because it's a class component and `useNavigate` is not available.

Reply:
`window.location.href` is used because in the first beginning we will run _coral_ embedded in the Klaw core project. 
When we build the bundle with [.env.springboot](https://github.com/aiven/klaw/blob/main/coral/.env.springboot) where we have `VITE_ROUTER_BASENAME=/coral/`. AFAIK if we would do `navigate("/login")` the navigation would actually take user to `/coral/login`, which does not exist as we want to use the AngularJS login.

It is also possible that `useNavigate` is not available in class components, but once we implement the login to _coral_ we should change this to use React Router.

Reply:
That makes absolute sense, thank you. Maybe we could have a comment there to remind us to swap this out once we implement login in coral?